pkg/resource/deploy/step.go

```diff
@@ -705,6 +706,15 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 	}
 
 	if outputs != nil {
+		// There is a chance that the ID has changed. We want to allow this change to happen
+		// it will have changed already in the outputs, but we need to persist this change
+		// at a state level because the Id
+		refreshedID := getIDFromOutputs(outputs)
```

Comment:
The `id` should already be a part of the `Read` response. Let's pull it from there.